great_expectations/types/__init__.py

```diff
@@ -19,6 +22,44 @@ def __contains__(self, key):
     def __len__(self):
         return len(self.__dict__)
 
+    def keys(self) -> set:
+        return self.to_dict().keys()
+
+    def items(self):
+        return self.to_dict().items()
+
+    def get(self, key, default_value=None):
+        return self.__dict__.get(key, default_value)
+
+    def to_dict(self):
+        new_dict = copy.deepcopy(self.__dict__)
+
+        # This is needed to play nice with pydantic.
+        if "__initialised__" in new_dict:
+            del new_dict["__initialised__"]
+
+        # DictDot's to_dict method works recursively, when a DictDot contains other DictDots.
+        for key, value in new_dict.items():
+            # Recursive conversion works on keys that are DictDots...
+            if isinstance(value, DictDot):
+                new_dict[key] = value.to_dict()
```

Comment:
do we need this method to be recursive and start with a deepcopy? Seems like we'd be creating a lot of copies of things

Reply:
I added a docstring to answer this question. Practically speaking, this makes a HUGE difference when trying to progressively type untyped parts of the codebase.